src/registrar/assets/src/js/getgov/table-edit-member-domains.js

```diff
@@ -217,7 +228,125 @@ export class EditMemberDomainsTable extends BaseTable {
+      unassignedHeader.classList.add('header--body', 'text-primary', 'margin-bottom-1');
+      unassignedHeader.textContent = 'Unassigned domains';
+      domainAssignmentSummary.appendChild(unassignedHeader);
+      domainAssignmentSummary.appendChild(unassignedDomainsList);
+    }
+
+    // Append assigned domains section
+    if (this.addedDomains.length) {
+      const assignedHeader = document.createElement('h3');
+      assignedHeader.classList.add('header--body', 'text-primary', 'margin-bottom-1');
+      assignedHeader.textContent = 'Assigned domains';
+      domainAssignmentSummary.appendChild(assignedHeader);
+      domainAssignmentSummary.appendChild(assignedDomainsList);
+    }
+
+    // Append total assigned domains section
+    const totalHeader = document.createElement('h3');
+    totalHeader.classList.add('header--body', 'text-primary', 'margin-bottom-1');
+    totalHeader.textContent = 'Total assigned domains';
+    domainAssignmentSummary.appendChild(totalHeader);
+    const totalCount = document.createElement('p');
+    totalCount.classList.add('margin-y-0');
+    totalCount.textContent = totalAssignedDomains;
+    domainAssignmentSummary.appendChild(totalCount);
+  }
+
+  showReadonlyMode() {
+    this.updateReadonlyDisplay();
+    hideElement(this.editModeContainer);
+    showElement(this.readonlyModeContainer);
+  }
+
+  showEditMode() {
+    hideElement(this.readonlyModeContainer);
+    showElement(this.editModeContainer);
+  }
+
+  submitChanges() {
+    let memberDomainsEditForm = document.getElementById("member-domains-edit-form");
+    if (memberDomainsEditForm) {
+      // Serialize data to send
+      const addedDomainIds = this.addedDomains.map(domain => domain.id);
+      const addedDomainsInput = document.createElement('input');
+      addedDomainsInput.type = 'hidden';
+      addedDomainsInput.name = 'added_domains'; // Backend will use this key to retrieve data
+      addedDomainsInput.value = JSON.stringify(addedDomainIds); // Stringify the array
+      
+      const removedDomainsIds = this.removedDomains.map(domain => domain.id);
+      const removedDomainsInput = document.createElement('input');
+      removedDomainsInput.type = 'hidden';
+      removedDomainsInput.name = 'removed_domains'; // Backend will use this key to retrieve data
+      removedDomainsInput.value = JSON.stringify(removedDomainsIds); // Stringify the array
+
+      // Append input to the form
+      memberDomainsEditForm.appendChild(addedDomainsInput);
+      memberDomainsEditForm.appendChild(removedDomainsInput);
+
+      memberDomainsEditForm.submit();
+    }
+  }
+
+  initEventListeners() {
+    if (this.reviewButton) {
+      this.reviewButton.addEventListener('click', () => {
+        this.showReadonlyMode();
+      });
+    } else {
+      console.warn('Missing DOM element. Expected element with id review-domain-assignments')
+    }
+
+    if (this.backButton.length == 2) {
```

Comment:
**non blocking question (also happy to just huddle if chatting is faster)** this is the error I was seeing, can you explain what's happening here a bit more?

Reply:
Yeah, we started adding JS console warnings when we make a string assumption that an element with a certain ID/class exists and we add some event listeners to it. If that element does not exist, the JS will error out and the stack will stop running. The console warnings basically do the job of a try/catch

We only started trying this out, not sure whether this is a practice that's good enough to adopt everywhere we're adding event listeners or doing actions on DOM elements.

Reply:
That makes sense, resolving.